README.md

```diff
@@ -67,12 +67,15 @@ The `UMF_BUILD_LIBUMF_POOL_SCALABLE` option has to be turned `ON` to build this
 ### Requirements
 
 Required packages:
-- C++ compiler with C++17 support
+- C compiler
```

Comment:
not sure how much you want to rely on this info... perhaps we should have a test for that? at least you can verify one time on the OS with no C++ compiler (pls send a link to the build, if go this way 😉 )

Reply:
I've made a [test branch](https://github.com/PatKamin/unified-memory-framework/actions/runs/7625835325/job/20770973393) to run CI without a C++ compiler and it looks like the project won't compile because the `project` CMake function fails when no C++ compiler is detected.
So currently the UMF won't be compiled just because the current form of `project()` include implicit checks both for C and C++ compilers.

I've added an explicit C-only CMake setup in cases where none of the two flags requiring C++ is set. This will help in keeping the bulk of UMF a pure C code.